Tasks/FtpUploadV2/ftpuploadtask.ts

```diff
@@ -23,142 +20,322 @@ export class FtpOptions {
     // advanced options
     clean: boolean;
     cleanContents: boolean;
-    overwrite: boolean; 
-    preservePaths: boolean; 
-    trustSSL: boolean; 
+    preservePaths: boolean;
+    trustSSL: boolean;
+}
+
+class ProgressTracking {
```

Comment:
Is there any reason why visibility markers aren't here?  e.g. `public` and `private` for the method names and the member variables?

Reply:
nit: `ProgressTracker`

Reply:
Yeah, it should have one.  I merged this from another file before, forgot to add a qualifier.

Reply:
I can't add any modifier, it complains 
```
'private' modifier cannot appear on a module or namespace element.ts(1044)
```
so I think this is fine, it's not exported anyway.